build-tools/packages/build-cli/src/lib/git.ts

```diff
@@ -193,14 +193,17 @@ export class Repository {
 			.filter((value) => value !== null && value !== undefined && value !== "");
 	}
 
-	public async canMergeWithoutConflicts(commit: string): Promise<boolean> {
+	public async canMergeWithoutConflicts(commit: string, creds: string[]): Promise<boolean> {
 		let mergeResult;
+		await this.git.addConfig("user.email", creds[0]);
+		await this.git.addConfig("user.name", creds[1]);
```

Comment:
I would not do this here. This function shouldn't need to configure the repo. Instead, this function should assume that these config settings have been set. Then you can just set them once in the run method and not pass the creds around - you just use the flags to set the repo settings once.

Reply:
Setting them in the GitHub workflow